File: frontend/lib/api/models/user_model.dart
class UserModel {
  final int? id;
  final String? schoolID;
  final String? email;
  final String? firstName;
  final String? middleName;
  final String? lastName;
  final String? profilePicture;
  final String? gender;
  final String? password;
  final String? contact;
  final String? address;
  final String? role;
  final String? professor;
  final String? profId;
  final String? hkType;
  final String? status;
  final String? token;

  UserModel({
    this.id,
    this.schoolID,
    this.email,
    this.firstName,
    this.middleName,
    this.lastName,
    this.profilePicture,
    this.gender,
    this.password,
    this.contact,
    this.address,
    this.role,
    this.professor,
    this.profId,
    this.hkType,
    this.status,
    this.token,
  });

  factory UserModel.fromJson(Map<String, dynamic> map) {
    return UserModel(
      schoolID: map['school_id'] ?? '',
      email: map['email'] ?? '',
      firstName: map['first_name'] ?? '',
      middleName: map['middle_name'] ?? '',
      lastName: map['last_name'] ?? '',
      profilePicture: map['profile_picture'] ?? '',
      gender: map['gender'] ?? '',
      password: map['password'] ?? '',
      contact: map['contact'] ?? '',
      address: map['address'] ?? '',
      role: map['role'] ?? '',
      professor: map['professor'] ?? '',
      profId: map['prof_id'] ?? '',
      hkType: map['hk_type'],
      status: map['status'] ?? '',
      token: map['token'] ?? '',
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'school_id': schoolID,
      'email': email,
      'first_name': firstName,
      'middle_name': middleName,
      'last_name': lastName,
      'profile_picture': profilePicture,
      'gender': gender,
      'password': password,
      'contact': contact,
      'address': address,
      'role': role,
      'professor': professor,
      'prof_id': profId,
      'hk_type': hkType,
      'status': status,
      'token': token,
    };
  }

  @override
  String toString() {
    return 'UserModel(id: $id, schoolID: $schoolID, email: $email, firstName: $firstName, middleName: $middleName, lastName: $lastName, role: $role, professor: $professor, profId: $profId,status: $status)';
  }
}

File: backend/models/user_model.js
const mongoose = require('mongoose');

const userSchema = new mongoose.Schema({
  school_id: { type: String, required: true, index: { unique: true} },
  email: { type: String, required: true, index: true },
  password: { type: String, required: true },
  first_name: { type: String, required: true },
  middle_name: { type: String, required: false },
  last_name: { type: String, required: true },
  profile_picture: { type: String, default: null }, 
  gender: { type: String, required: true },
  contact: { type: String, required: true },
  address: { type: String, required: true },
  role: { type: String, required: true },
  professor: { type: String, required: false, default: null},
  hk_type: { type: String, required: false },
  status: { type: String, required: true },
  token: { type: String, required: false }
}, { timestamps: true });

const User = mongoose.model('User', userSchema, 'users');

module.exports = User;
